nativelib/src/main/scala/scala/scalanative/posix/pthread.scala

```diff
@@ -0,0 +1,365 @@
+
+  def pthread_mutexattr_setpshared(attr: Ptr[pthread_mutexattr_t],
+                                   pshared: CInt): CInt = extern
+
+  def pthread_mutexattr_setrobust(attr: Ptr[pthread_mutexattr_t],
+                                  robust: CInt): CInt = extern
+
+  def pthread_mutexattr_settype(attr: Ptr[pthread_mutexattr_t],
+                                tp: CInt): CInt = extern
+
+  def pthread_once(once_control: Ptr[pthread_once_t],
+                   init_routine: CFunctionPtr0[Unit]): CInt = extern
+
+  def pthread_rwlock_destroy(rwlock: Ptr[pthread_rwlock_t]): CInt = extern
+
+  def pthread_rwlock_init(rwlock: Ptr[pthread_rwlock_t],
+                          attr: Ptr[pthread_rwlockattr_t]): CInt = extern
+
+  def pthread_rwlock_rdlock(rwlock: Ptr[pthread_rwlock_t]): CInt = extern
+
+  def pthread_rwlock_timedrdlock(rwlock: Ptr[pthread_rwlock_t],
+                                 timespec: Ptr[timespec]): CInt = extern
+
+  def pthread_rwlock_timedwrlock(rwlock: Ptr[pthread_rwlock_t],
+                                 timespec: Ptr[timespec]): CInt = extern
+
+  def pthread_rwlock_tryrdlock(rwlock: Ptr[pthread_rwlock_t]): CInt = extern
+
+  def pthread_rwlock_trywrlock(rwlock: Ptr[pthread_rwlock_t]): CInt = extern
+
+  def pthread_rwlock_unlock(rwlock: Ptr[pthread_rwlock_t]): CInt = extern
+
+  def pthread_rwlock_wrlock(rwlock: Ptr[pthread_rwlock_t]): CInt = extern
+
+  def pthread_rwlockattr_destroy(attr: Ptr[pthread_rwlockattr_t]): CInt =
+    extern
+
+  def pthread_rwlockattr_getpshared(attr: Ptr[pthread_rwlockattr_t],
+                                    pshared: Ptr[CInt]): CInt = extern
+
+  def pthread_rwlockattr_init(attr: Ptr[pthread_rwlockattr_t]): CInt = extern
+
+  def pthread_rwlockattr_setpshared(attr: Ptr[pthread_rwlockattr_t],
+                                    pshared: CInt): CInt = extern
+
+  def pthread_self(): pthread_t = extern
+
+  def pthread_setcancelstate(state: CInt, oldstate: Ptr[CInt]): CInt = extern
+
+  def pthread_setcanceltype(tp: CInt, oldtype: Ptr[CInt]): CInt = extern
+
+  def pthread_setconcurrency(concurrency: CInt): CInt = extern
+
+  def pthread_setschedparam(thread: pthread_t,
+                            policy: CInt,
+                            param: Ptr[sched_param]): CInt = extern
+
+  def pthread_setschedprio(thread: pthread_t, schedprio: CInt): CInt = extern
+
+  def pthread_setspecific(key: pthread_key_t, value: Ptr[Byte]): CInt = extern
+
+  def pthread_spin_destroy(spinlock: Ptr[pthread_spinlock_t]): CInt = extern
+
+  def pthread_spin_init(spinlock: Ptr[pthread_spinlock_t],
+                        pshared: CInt): CInt = extern
+
+  def pthread_spin_lock(spinlock: Ptr[pthread_spinlock_t]): CInt = extern
+
+  def pthread_spin_trylock(spinlock: Ptr[pthread_spinlock_t]): CInt = extern
+
+  def pthread_spin_unlock(spinlock: Ptr[pthread_spinlock_t]): CInt = extern
+
+  def pthread_testcancel(): Unit = extern
+
+  // Macros
```

Comment:
@Korf74 You can also look at `grp.scala` to see how to create `type`s to make the API more readable as well. All the code in that package are good to look at to get ideas. Good progress. https://github.com/scala-native/scala-native/search?utf8=%E2%9C%93&q=grp&type=